instrumentation/kafka/kafka-connect-2.6/testing/src/test/java/io/opentelemetry/instrumentation/kafkaconnect/v2_6/MongoKafkaConnectSinkTaskTest.java

```diff
@@ -0,0 +1,354 @@
+    String testTopicName = TOPIC_NAME;
+    setupMongoSinkConnector(testTopicName);
+
+    createTopic(testTopicName);
+    awaitForTopicCreation(testTopicName);
+
+    Properties props = new Properties();
+    props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, getKafkaBoostrapServers());
+    props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+    props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+
+    testing.waitForTraces(9); // Skip initial traces from Kafka Connect startup
+    testing.clearData();
+
+    try (Producer<String, String> producer = instrument(new KafkaProducer<>(props))) {
+      producer.send(
+          new ProducerRecord<>(testTopicName, "test-key", "{\"id\":1,\"name\":\"TestUser\"}"));
+      producer.flush();
+    }
+
+    await().atMost(Duration.ofSeconds(60)).until(() -> getRecordCountFromMongo() >= 1);
+
+    AtomicReference<SpanContext> producerSpanContext = new AtomicReference<>();
+    testing.waitAndAssertTraces(
+        trace ->
+            // producer is in a separate trace, linked to consumer with a span link
+            trace.hasSpansSatisfyingExactly(
+                span -> {
+                  span.hasName(testTopicName + " publish").hasKind(SpanKind.PRODUCER).hasNoParent();
+                  producerSpanContext.set(span.actual().getSpanContext());
+                }),
+        trace ->
+            // kafka connect sends message to status topic while processing our message
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("kafka-connect-status publish")
+                        .hasKind(SpanKind.PRODUCER)
+                        .hasNoParent(),
+                span ->
+                    span.hasName("kafka-connect-status process")
+                        .hasKind(SpanKind.CONSUMER)
+                        .hasParent(trace.getSpan(0))),
+        trace ->
+            // kafka connect consumer trace, linked to producer span via a span link
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName(testTopicName + " process")
+                        .hasKind(CONSUMER)
+                        .hasNoParent()
+                        .hasLinks(LinkData.create(producerSpanContext.get())),
+                span ->
+                    span.hasName("update " + DB_NAME + "." + COLLECTION_NAME)
+                        .hasKind(SpanKind.CLIENT)
+                        .hasParent(trace.getSpan(0))),
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span -> span.hasName("GET /connectors").hasKind(SpanKind.SERVER).hasNoParent()),
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span -> span.hasName("GET /connectors").hasKind(SpanKind.SERVER).hasNoParent()));
+  }
+
+  @Test
+  public void testKafkaConnectMongoSinkTaskMultiTopicInstrumentation() throws Exception {
```

Comment:
this note applies to all test methods, methods don't need to be public

```suggestion
  void testKafkaConnectMongoSinkTaskMultiTopicInstrumentation() throws Exception {
```